File: enqueue.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 5

//for arrays
int queue[SIZE];
int front_arr = -1, rear_arr = -1;

void enqueue_array(int value) {
    if (rear_arr == SIZE - 1) {
        printf("Queue is full! %d\n", value);
    } else {
        if (front_arr == -1) front_arr = 0;
        queue[++rear_arr] = value;
        printf("Added %d to array queue.\n", value);
    }
}

//for linked lists
struct Node {
    int data;
    struct Node* next;
};

struct Node* front_list = NULL;
struct Node* rear_list = NULL;

void enqueue_list(int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;

    if (rear_list == NULL) {
        front_list = rear_list = newNode;
    } else {
        rear_list->next = newNode;
        rear_list = newNode;
    }
    printf("%d enqueued into linked list queue.\n", value);
}

int main() {
    printf("in arrays.\n");
    enqueue_array(10);
    enqueue_array(20);
    enqueue_array(30);
    printf("in linked lists.\n");
    enqueue_list(3);
    enqueue_list(2);
    enqueue_list(1);
    return 0;
}

File: enqueue.java
public class enqueue {
//for arrays
static int SIZE = 5;
static int[] queueA = new int[SIZE];
static int frontA = -1, rearA = -1;

static void enqueueArray(int value) {
    if (rearA == SIZE - 1) {
        System.out.println("Array Queue is FULL!");
        return;
    }
    if (frontA == -1) frontA = 0;
    queueA[++rearA] = value;
    System.out.println("Enqueued (Array): " + value);
}

//for linked lists
static class Node {
    int data;
    Node next;
    Node(int val) { data = val; next = null; }
}

static Node frontL = null, rearL = null;

static void enqueueLinkedList(int value) {
    Node newNode = new Node(value);
    if (rearL == null)
        frontL = rearL = newNode;
    else {
        rearL.next = newNode;
        rearL = newNode;
    }
    System.out.println("Enqueued (Linked List): " + value);
}

public static void main(String[] args) {
    System.out.println("Using Array Queue:");
    enqueueArray(10);
    enqueueArray(20);
    enqueueArray(30);
    enqueueArray(40);
    enqueueArray(50);
    enqueueArray(60); // This should trigger "Array Queue is FULL!"

    System.out.println("\nUsing Linked List Queue:");
    enqueueLinkedList(30);
    enqueueLinkedList(40);
    enqueueLinkedList(50);
    enqueueLinkedList(60);
}
}
